modules/drivers/mongo/test/metabase/driver/mongo/parameters_test.clj

```diff
@@ -78,25 +80,71 @@
            (substitute {:id (comma-separated-numbers [1 2 3])}
                        [(param :id)])))))
 
+(defprotocol ^:private ToBSON
+  (^:private to-bson [this]
+   "Utility method for converting normal Clojure objects to string-serialized 'BSON'."))
+
+(extend-protocol ToBSON
+  nil
+  (to-bson [_] "null")
+
+  Object
+  (to-bson [this] (pr-str this))
+
+  clojure.lang.Keyword
+  (to-bson [this] (name this))
+
+  clojure.lang.Sequential
+  (to-bson [this]
+    (str "["
+         (str/join ", " (map to-bson this))
+         "]"))
+
+  clojure.lang.IPersistentMap
+  (to-bson [this]
+    (str "{"
+         (str/join ", " (for [[k v] this]
+                          (str (to-bson k) ": " (to-bson v))))
+         "}")))
+
+(defn- bson-fn-call [f & args]
+  (reify ToBSON
+    (to-bson [_]
+      (format "%s(%s)" (name f) (str/join "," (map pr-str args))))))
+
+(defn- ISODate [s]
+  (bson-fn-call :ISODate s))
+
 (deftest field-filter-test
   (testing "Date ranges"
     (mt/with-clock #t "2019-12-13T12:00:00.000Z[UTC]"
-      (is (= "[{$match: {$and: [{\"date\": {$gte: ISODate(\"2019-12-08\")}}, {\"date\": {$lt: ISODate(\"2019-12-12\")}}]}}]"
-             (substitute {:date (field-filter "date" :date/range "past5days")}
-                         ["[{$match: " (param :date) "}]"])))))
+      (letfn [(substitute-date-range [s]
+                (substitute {:date (field-filter "date" :date/range s)}
+                            ["[{$match: " (param :date) "}]"]))]
+        (is (= (to-bson [{:$match {:$and [{"date" {:$gte (ISODate "2019-12-08")}}
```

Comment:
reading big serialized BSON strings was hurting my head so I slapped together a quick BSON serializer to make the tests easier to read